data/secscan_model/interface.py

```diff
@@ -0,0 +1,36 @@
+from abc import ABCMeta, abstractmethod, abstractproperty
+from six import add_metaclass
+
+
+@add_metaclass(ABCMeta)
+class SecurityScannerInterface(object):
+    """ Interface for code to work with the security scan data model. This model encapsulates
+        all access when speaking to an external security scanner, as well as any data tracking
+        in the database.
+    """
+
+    @abstractmethod
+    def load_security_information(self, manifest_or_legacy_image, include_vulnerabilities=False):
+        """ Loads the security information for the given manifest or legacy image, returning
+            a SecurityInformationLookupResult structure. The manifest_or_legacy_image must be a Manifest
+            or LegacyImage datatype from the registry_model.
+        """
+
+    @abstractmethod
+    def perform_indexing(self, start_token=None):
+        """ Performs indexing of the next set of unindexed manifests/images. If start_token is given,
+            the indexing should resume from that point. Returns a new start index for the next
+            iteration of indexing. The tokens returned and given are assumed to be opaque outside
+            of this implementation and should not be relied upon by the caller to conform to any
+            particular format.
+        """
+
+    @abstractmethod
+    def register_model_cleanup_callbacks(self, data_model_config):
```

Comment:
I don't know the background of this, but it looks like there is only only ever used once in `app.py` for `secscan_model.register_model_cleanup_callbacks(model.config)`. But I can't find any other uses of this [callback method registration in `data/model/__init__.py`](https://github.com/quay/quay/blob/master/data/model/__init__.py#L117).

Reply:
Correct, but there are other callbacks too, and I didn't want to too tightly couple them